cpp/test/sg/fil_test.cu

```diff
@@ -400,36 +413,60 @@ INSTANTIATE_TEST_CASE_P(FilTests, PredictFilTest,
+   2e-3f, tl::Operator::kGE},
+  {20000, 50, 0.05, 8, 50, 0.05,
+   fil::output_t(fil::output_t::AVG | fil::output_t::THRESHOLD), 0,
+   fil::algo_t::NAIVE, 42, 2e-3f, tl::Operator::kLT},
   {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::RAW, 0, fil::algo_t::TREE_REORG,
-   42, 2e-3f, tl::Operator::kGE},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::PROB, 0,
-   fil::algo_t::TREE_REORG, 42, 2e-3f, tl::Operator::kLT},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::CLASS, 0,
+   42, 2e-3f, tl::Operator::kLE},
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::SIGMOID, 0,
+   fil::algo_t::TREE_REORG, 42, 2e-3f, tl::Operator::kGT},
+  {20000, 50, 0.05, 8, 50, 0.05,
+   fil::output_t(fil::output_t::SIGMOID | fil::output_t::THRESHOLD), 0,
+   fil::algo_t::TREE_REORG, 42, 2e-3f, tl::Operator::kGE},
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::AVG, 0, fil::algo_t::TREE_REORG,
+   42, 2e-3f, tl::Operator::kLT},
+  {20000, 50, 0.05, 8, 50, 0.05,
+   fil::output_t(fil::output_t::AVG | fil::output_t::THRESHOLD), 0,
    fil::algo_t::TREE_REORG, 42, 2e-3f, tl::Operator::kLE},
   {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::RAW, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kLT},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::PROB, 0,
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::SIGMOID, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kLT},
   {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::RAW, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kLE},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::PROB, 0,
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::SIGMOID, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kLE},
   {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::RAW, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kGT},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::PROB, 0,
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::SIGMOID, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kGT},
   {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::RAW, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kGE},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::PROB, 0,
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::SIGMOID, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kGE},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::CLASS, 0,
+  {20000, 50, 0.05, 8, 50, 0.05,
+   fil::output_t(fil::output_t::SIGMOID | fil::output_t::THRESHOLD), 0,
+   fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kLT},
+  {20000, 50, 0.05, 8, 50, 0.05,
+   fil::output_t(fil::output_t::SIGMOID | fil::output_t::THRESHOLD), 0,
+   fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kLE},
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::AVG, 0,
    fil::algo_t::BATCH_TREE_REORG, 42, 2e-3f, tl::Operator::kLT},
-  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::CLASS, 0,
+  {20000, 50, 0.05, 8, 50, 0.05, fil::output_t::AVG, 0,
```

Comment:
Have these tests become pretty slow now? Or still a matter of seconds? Nice to have a bunch of them but we may want to reduce depth and nsamples as this list grows.